client/src/reducers/controls.js

```diff
@@ -128,6 +134,16 @@ const Controls = (
           );
         }
       });
+
+      _.forEach(action.expressionData, (val, key) => {
+        dimensionMap[key] = World.createVarDimension(
```

Comment:
Note that the current implementation will 'lose' all of these dimensions whenever you regraph (set world to the current selection).   The gene expression histograms will still be displayed (because you are displaying a histogram for anything in world.varDataCache), so if the user brushes the histogram, they will get a hard error.

General rule of thumb:
- dimensionMap needs to be kept in sync with world
- it also needs to be kept in sync with whatever state you use to store the list of genes to display a histogram for (currently, world.varDataCache).

Currently, the only actions that mess with this are:
* `reset World to eq Universe`
* `set World to current selection`

These rebuild the obs annotation dimensionMap when world changes.   If you want to keep things consistent, you need to rebuild the var data dimensions as well.